hudi-common/src/main/java/org/apache/hudi/common/bootstrap/index/HFileBootstrapIndex.java

```diff
@@ -240,13 +240,21 @@ private HoodieBootstrapIndexInfo fetchBootstrapIndexInfo() throws IOException {
     @Override
     public List<String> getIndexedPartitionPaths() {
       HFileScanner scanner = partitionIndexReader().getScanner(true, true);
-      return getAllKeys(scanner);
+      List<String> cellKeys = getAllKeys(scanner);
+      // cellKey is in this format:
+      // part=datestr=2452537//LATEST_TIMESTAMP/Put/vlen=2405/seqid=0
+      return cellKeys.stream().map(key -> key.split("//")[0].substring(5))
+          .distinct().collect(Collectors.toList());
     }
 
     @Override
     public List<String> getIndexedFileIds() {
       HFileScanner scanner = fileIdIndexReader().getScanner(true, true);
-      return getAllKeys(scanner);
+      List<String> cellKeys = getAllKeys(scanner);
+      // cellKey is in this format:
+      // part=datestr=2452537;fileid=baab9c50-c35e-49d1-b928-695aa7e37833//LATEST_TIMESTAMP/Put/vlen=2312/seqid=0
+      return cellKeys.stream().map(key -> key.split("//")[0].split(";")[1].split("=")[1])
```

Comment:
were the values currently returning something incorrect?  if so, how were the tests happy?

Reply:
I have this question too, but IIUC this code path was probably not being used till now. @zhedoubushishi can confirm.

Reply:
These two methods are only called in ```TestBootstrapIndex```. And what it does in the test code is to check the size of the result, so that's why we fail to catch this error before. Let me create a separate commit/pr to include this fix and also add some test cases to cover these two methods.

Reply:
This is a known issue: https://issues.apache.org/jira/browse/HUDI-971

Reply:
@zhedoubushishi :  Can you instead copy the code from the PR https://github.com/apache/hudi/pull/1933. This is essentially I fixed. I did not realize you have tried to address the issue here. IMO, PR 1933 is more cleaner

Reply:
@zhedoubushishi : I will assign the original ticket to you

Reply:
@bvaradar thanks! PR #1933 looks good to me. Migrated it into current PR.